src/classes/OPP_CampaignMember_TDTM.cls

```diff
@@ -58,7 +58,7 @@ public class OPP_CampaignMember_TDTM extends TDTM_Runnable {
         
         dmlWrapper dmlWrapper = new DmlWrapper();
 
-        if (hasRun) {
+        if (recursionFlag > 0) {
```

Comment:
I think there is benefit in being consistent and following our new process control paradigm.  that said, you could create the process control class as an inner class private to just this outer class.